### PR TITLE
Base Components / React-Router

### LINKED ISSUE #38: Add Page and Component Libraries and Integrate React-Router

### PR DESCRIPTION
**Pull request by tmurphy3**
## Related Issues (include the `#`)
#38 

## Description of Changes Made
Added react-router, added home component, sidebar component, and navbar component.

### CODE REVIEW

File: front-end/src/App.js [hunk 1]
```diff
@@ -1,9 +1,29 @@
-import React from 'react';
+import React from "react";
+import Profile from "./Components/Profile.js";
+import Home from "./Components/Home.js";
+import Navbar from "./Components/Navbar.js";
+import Sidebar from "./Components/Sidebar.js";
```
Comment (midastouchprd) [inline front-end/src/App.js]: lets seperate the idea of components and pages

Navbar and Sidebar are components
Profile and Home are pages

so you should probably make a pages directory and a components directory.

File: front-end/src/Components/Projects.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+
+function Projects() {
```
Comment (midastouchprd) [inline front-end/src/Components/Projects.js]: lets leave this in components, but change the name to ProjectsList.

because the could be a Project Viewer, or a Project Grader, or a Project Dashboard so we wanna be specific

File: front-end/src/axios.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import axios from "axios";
+
+const instance = axios.create({
+  baseURL: "https://developer-verse.herokuapp.com/",
+});
```
Comment (midastouchprd) [inline front-end/src/axios.js]: this is good, but you'll want to change the base instance depending on your env

so if its in production that's the way to go, but when I'M developing I wanna be using `localhost`

**Review by midastouchprd — CHANGES_REQUESTED** [review]
Just a few minor things!

File: front-end/src/axios.js (hunk 3 above)
Comment (tmurphy3) [reply front-end/src/axios.js]: this is the deployed heroku app, are you saying i should be using my local backend database?

File: front-end/src/Components/Projects.js (hunk 2 above)
Comment (tmurphy3) [reply front-end/src/Components/Projects.js]: Okay, I just made these changes.

File: front-end/src/App.js (hunk 1 above)
Comment (tmurphy3) [reply front-end/src/App.js]: Okay, I just made these changes.

File: front-end/src/axios.js (hunk 3 above)
Comment (Underwaterr) [reply front-end/src/axios.js]: The source code should run with a different base URL depending on if it's running locally on a dev's machine or if it's running in the test environment. So you'll be implementing environment variables for the React app in this case.

Comment (midastouchprd) [reply front-end/src/axios.js]: it'll look like 

```
instance;

if (envvariable === "development) {
  instance = axios.create({baseURL: 'local})
} else {
  instance = axios.create({baseURL: 'deployed})
}

**Review by midastouchprd — CHANGES_REQUESTED** [review]
Just the ENV stuff and this is good!

File: back-end/developerverse/models.py [hunk 4]
```diff
@@ -1,3 +1,9 @@
 from django.db import models
+from django.contrib.auth.models import AbstractUser
 
 # Create your models here.
+class User(AbstractUser):
+    email = models.EmailField(unique=True)
+
+    USERNAME_FIELD = 'email'
+    REQUIRED_FIELDS = []
```
Comment (Underwaterr) [inline back-end/developerverse/models.py]: This file and others should not be part of the commit. Remove any unrelated back-end stuff from tracking otherwise we'll have a messy merge ahead.

**Review by Underwaterr — CHANGES_REQUESTED** [review]
Fix tracked files
